flink-table-store-kafka/src/main/java/org/apache/flink/table/store/kafka/KafkaLogStoreFactory.java

```diff
@@ -237,6 +241,14 @@ public KafkaLogSinkProvider createSinkProvider(
                 helper.getOptions().get(CHANGELOG_MODE));
     }
 
+    private int[] getPrimaryKeyIndexes(ResolvedSchema schema) {
+        final List<String> columns = schema.getColumnNames();
+        return schema.getPrimaryKey()
+                .map(UniqueConstraint::getColumns)
+                .map(pkColumns -> pkColumns.stream().mapToInt(columns::indexOf).toArray())
+                .orElseGet(() -> new int[] {});
+    }
+
```

Comment:
Just curious is there any difference between `getPrimaryKeyIndexes` and `schema.getPrimaryKeyIndexes`?

Reply:
1.14 dose not provide `schema.getPrimaryKeyIndexes`, just copy from 1.15.